### PR TITLE
[CALCITE-6474] Aggregate with constant key can get a RowCount greater than its MaxRowCount

### PR DESCRIPTION
**Pull request by rubenada**
More info: CALCITE-6474

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/RelMetadataTest.java [hunk 1]
```diff
@@ -788,6 +788,20 @@ final RelMetadataFixture sql(String sql) {
     sql(sql).assertThatRowCount(is(1D), is(0D), is(1D));
   }
 
+  @Test void testRowCountAggregateConstantKeysOnBigInput() {
```
Comment (mihaibudiu) [inline core/src/test/java/org/apache/calcite/test/RelMetadataTest.java:791]: It's not obvious why this is an aggregate, I assume some rewrite step converts it into one.
That could use a comment.
Do you want to mention the JIRA case?

**Review by mihaibudiu — APPROVED** [review]

File: core/src/main/java/org/apache/calcite/rel/metadata/RelMdMaxRowCount.java [hunk 2]
```diff
@@ -159,7 +159,7 @@ && allGroupKeysAreConstant(rel, predicateList)) {
     return rowCount * rel.getGroupSets().size();
   }
 
-  private static boolean allGroupKeysAreConstant(Aggregate aggregate,
+  static boolean allGroupKeysAreConstant(Aggregate aggregate,
```
Comment (caicancai) [inline core/src/main/java/org/apache/calcite/rel/metadata/RelMdMaxRowCount.java:162]: protected or public might be better?

File: core/src/test/java/org/apache/calcite/test/RelMetadataTest.java (hunk 1 above)
Comment (caicancai) [inline core/src/test/java/org/apache/calcite/test/RelMetadataTest.java:791]: Add jira link

Comment (caicancai) [reply core/src/test/java/org/apache/calcite/test/RelMetadataTest.java:791]: +1

File: core/src/main/java/org/apache/calcite/rel/metadata/RelMdMaxRowCount.java (hunk 2 above)
Comment (rubenada) [reply core/src/main/java/org/apache/calcite/rel/metadata/RelMdMaxRowCount.java:162]: We want to reuse this in RelMdRowCount, so I think it's better to have it package-private.

File: core/src/test/java/org/apache/calcite/test/RelMetadataTest.java (hunk 1 above)
Comment (rubenada) [reply core/src/test/java/org/apache/calcite/test/RelMetadataTest.java:794]: done

Comment (rubenada) [reply core/src/test/java/org/apache/calcite/test/RelMetadataTest.java:794]: I think it is arguably clear that the aggregate comes from the count (I'm just replicating the same patter of test method name and test code from the existing ones).
Added javadoc with Jira case.

**Review by caicancai — APPROVED** [review]
LGTM

**Pull request merged**